### ISSUE TITLE
failed to install spatial-correlation-sampler

### ISSUE THREAD

**Issue by rihns38** [ClementPinard/Pytorch-Correlation-extension#96]
Hi,

I am using:
_Windows 10
_Python3.9
_torch2.0
_cuda 12.1
_gcc13.1

When I try "pip install spatial-correlation-sampler", I get the following error:

Collecting git+https://github.com/ClementPinard/Pytorch-Correlation-extension.git
  Cloning https://github.com/ClementPinard/Pytorch-Correlation-extension.git to c:\users\lecia\appdata\local\temp\pip-req-build-_gotqngo
  Running command git clone --filter=blob:none --quiet https://github.com/ClementPinard/Pytorch-Correlation-extension.git 'C:\Users\Lecia\AppData\Local\Temp\pip-req-build-_gotqngo'
  Resolved https://github.com/ClementPinard/Pytorch-Correlation-extension.git to commit 41c8d376a002f2f2d14d2769e380c4d9809a296c
  Preparing metadata (setup.py) ... done
Requirement already satisfied: torch>=1.1 in c:\users\lecia\anaconda3\envs\aot\lib\site-packages (from spatial-correlation-sampler==0.4.0) (2.0.1+cu117)
Requirement already satisfied: numpy in c:\users\lecia\anaconda3\envs\aot\lib\site-packages (from spatial-correlation-sampler==0.4.0) (1.22.3)
Requirement already satisfied: filelock in c:\users\lecia\anaconda3\envs\aot\lib\site-packages (from torch>=1.1->spatial-correlation-sampler==0.4.0) (3.9.0)
Requirement already satisfied: sympy in c:\users\lecia\anaconda3\envs\aot\lib\site-packages (from torch>=1.1->spatial-correlation-sampler==0.4.0) (1.11.1)
Requirement already satisfied: jinja2 in c:\users\lecia\anaconda3\envs\aot\lib\site-packages (from torch>=1.1->spatial-correlation-sampler==0.4.0) (3.1.2)
Requirement already satisfied: typing-extensions in c:\users\lecia\anaconda3\envs\aot\lib\site-packages (from torch>=1.1->spatial-correlation-sampler==0.4.0) (4.4.0)
Requirement already satisfied: networkx in c:\users\lecia\anaconda3\envs\aot\lib\site-packages (from torch>=1.1->spatial-correlation-sampler==0.4.0) (3.0)
Requirement already satisfied: MarkupSafe>=2.0 in c:\users\lecia\anaconda3\envs\aot\lib\site-packages (from jinja2->torch>=1.1->spatial-correlation-sampler==0.4.0) (2.1.2)
Requirement already satisfied: mpmath>=0.19 in c:\users\lecia\anaconda3\envs\aot\lib\site-packages (from sympy->torch>=1.1->spatial-correlation-sampler==0.4.0) (1.2.1)
Building wheels for collected packages: spatial-correlation-sampler
  Building wheel for spatial-correlation-sampler (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [12 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-39
      creating build\lib.win-amd64-cpython-39\spatial_correlation_sampler
      copying Correlation_Module\spatial_correlation_sampler\spatial_correlation_sampler.py -> build\lib.win-amd64-cpython-39\spatial_correlation_sampler
      copying Correlation_Module\spatial_correlation_sampler\__init__.py -> build\lib.win-amd64-cpython-39\spatial_correlation_sampler
      running build_ext
      C:\Users\Lecia\anaconda3\envs\aot\lib\site-packages\torch\utils\cpp_extension.py:359: UserWarning: Error checking compiler version for cl: [WinError 2] The system cannot find the file specified
        warnings.warn(f'Error checking compiler version for {compiler}: {error}')
      error: [WinError 2] The system cannot find the file specified
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for spatial-correlation-sampler
  Running setup.py clean for spatial-correlation-sampler
Failed to build spatial-correlation-sampler
Installing collected packages: spatial-correlation-sampler
  Running setup.py install for spatial-correlation-sampler ... error
  error: subprocess-exited-with-error

  × Running setup.py install for spatial-correlation-sampler did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      running install
      C:\Users\Lecia\anaconda3\envs\aot\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-39
      creating build\lib.win-amd64-cpython-39\spatial_correlation_sampler
      copying Correlation_Module\spatial_correlation_sampler\spatial_correlation_sampler.py -> build\lib.win-amd64-cpython-39\spatial_correlation_sampler
      copying Correlation_Module\spatial_correlation_sampler\__init__.py -> build\lib.win-amd64-cpython-39\spatial_correlation_sampler
      running build_ext
      C:\Users\Lecia\anaconda3\envs\aot\lib\site-packages\torch\utils\cpp_extension.py:359: UserWarning: Error checking compiler version for cl: [WinError 2] The system cannot find the file specified
        warnings.warn(f'Error checking compiler version for {compiler}: {error}')
      error: [WinError 2] The system cannot find the file specified
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> spatial-correlation-sampler

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

Please let me know if I'm doin something wrong here.

Thanks,
Renaud

**Comment by ClementPinard** [comment]
Hi, sorry this was specifically designed to be built on Linux.

I have no idea how to make it compatible with windows, and don't have acces to a computer with windows AND cuda.

Feel free to contribute to this project if you finally find out how to compile it on windows !

**Comment by JucaNavazReque** [comment]
> Hi, sorry this was specifically designed to be built on Linux.
> 
> I have no idea how to make it compatible with windows, and don't have acces to a computer with windows AND cuda.
> 
> Feel free to contribute to this project if you finally find out how to compile it on windows !

Please, specify this on the README. I lost way more time than I should have trying to make it run on Windows.

**Comment by ClementPinard** [comment]
No need to be rude here. This module is only using pytorch build tools, so in theory you should not have to change anything. But this was not tested on Windows and that's it.

If you want to build it for windows, don't hesitate to post about your attempts and findings and I will try to help as much as I can. If you're just here to rant, then congratulations, because then everybody lost their time.

**Comment by JucaNavazReque** [comment]
I wasn't trying to be rude, I'm really sorry if you interpreted it that way.

The only findings I got were several errors that lead me to this issue on your repository. I was trying to use a different project that has your library as a dependency, so it took me quite some time to realize what was going on. I lack on the knowledge needed to proceed any further. I just gave up trying to run the project on Windows, since probably more errors will appear after I solve the issue with spatial-correlation-sampler, as the project does not seem to be intended for Windows either.

My comment was more a suggestion than a rant. I tried several things that did not work and it took some time to get to this issue on your repo. I think it can be helpful for others if you specify that the project "was specifically designed to be built on Linux", and prevent them from taking the same time that I took to get here.

Again, sorry for the tone misunderstanding, and thank you for the quick response. Kind regards.
